File: PopularShows/app/src/main/java/com/lgcode/popularshows/TvShow.java
package com.lgcode.popularshows;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

/**
 * Created by leojg on 11/5/16.
 */
public class TvShow implements Serializable {

    private int id;
    @SerializedName("name") private String title;
    private String overview;
    @SerializedName("poster_path") private String image;
    @SerializedName("backdrop_path") private String heroImage;
    @SerializedName("vote_average") private Double voteAverage;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getOverview() {
        return overview;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public String getHeroImage() {
        return heroImage;
    }

    public void setHeroImage(String heroImage) {
        this.heroImage = heroImage;
    }

    public Double getVoteAverage() {
        return voteAverage;
    }
}

File: KtPopularShows/app/src/main/java/example/lgcode/ktpopularshows/domain/TVShow.kt
package example.lgcode.ktpopularshows.domain

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName

data class TVShow(
    val id: Int,
    @SerializedName("name")
    val title: String,
    val overview: String,
    @SerializedName("poster_path")
    val image: String,
    @SerializedName("backdrop_path")
    val heroImage: String,
    @SerializedName("vote_average")
    val voteAverage: Float
) : Parcelable {

    constructor(parcel: Parcel) : this(
            parcel.readInt(),
            parcel.readString(),
            parcel.readString(),
            parcel.readString(),
            parcel.readString(),
            parcel.readFloat()) {
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeInt(id)
        parcel.writeString(title)
        parcel.writeString(overview)
        parcel.writeString(image)
        parcel.writeString(heroImage)
        parcel.writeFloat(voteAverage)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<TVShow> {
        override fun createFromParcel(parcel: Parcel): TVShow {
            return TVShow(parcel)
        }

        override fun newArray(size: Int): Array<TVShow?> {
            return arrayOfNulls(size)
        }
    }
}